### PR TITLE
Integration test for ARM

### PR DESCRIPTION
**Pull request by eQu1NoX**

### CODE REVIEW

**Comment by eQu1NoX** [comment]
Tests work locally. Can be merged.

**Comment by zachriggle** [comment]
Can't be merged until it passes CI

**Comment by eQu1NoX** [comment]
Error log says : "OSError: [Errno 8] Exec format error"

**Comment by eQu1NoX** [comment]
"OSError: [Errno 8] Exec format error"

**Comment by zachriggle** [comment]
See the other thread :)

File: pwnlib/shellcraft/templates/arm/linux/read.asm [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<%docstring>Read 1024 bytes to sp from STDIN</%docstring>
```
Comment (zachriggle) [inline pwnlib/shellcraft/templates/arm/linux/read.asm]: This needs arguments, and should use the `syscall` template.

File: pwnlib/shellcraft/templates/arm/linux/sigreturn.asm [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<%docstring>Sigreturn system call</%docstring>
```
Comment (zachriggle) [inline pwnlib/shellcraft/templates/arm/linux/sigreturn.asm]: This should use the `constants` module

File: pwnlib/rop/srop.py [hunk 3]
```diff
@@ -79,7 +78,32 @@
     >>> p.wait_for_close()
     >>> p.poll() == 0
     True
+
+arm Example:
+
+    >>> context.clear()
+    >>> context.arch = "arm"
+    >>> assembly =  'read:'      + shellcraft.read()
```
Comment (zachriggle) [inline pwnlib/rop/srop.py]: This will need arguments after you make the changes to `read.asm`

**Comment by zachriggle** [comment]
Regarding my comments on arguments and documentation, see the shellcode template tutorial
https://github.com/binjitsu/tutorial/blob/master/walkthrough/shellcode-advanced/README.md

File: pwnlib/shellcraft/templates/arm/linux/read.asm (hunk 1 above)
Comment (eQu1NoX) [inline pwnlib/shellcraft/templates/arm/linux/read.asm]: syscall.asm uses "swi #0" ; for tests we need "svc <syscall-no>". change syscall.asm?

Comment (zachriggle) [inline pwnlib/shellcraft/templates/arm/linux/read.asm]: I'd recommend re-investigating the reason for that, it should work fine either way

Comment (eQu1NoX) [inline pwnlib/shellcraft/templates/arm/linux/read.asm]: The way I understand it : ABI difference in qemu-user / qemu-system.

Comment (eQu1NoX) [inline pwnlib/shellcraft/templates/arm/linux/read.asm]: wanna support both abi's ? that'd be nice.

**Pull request closed without merging**